### PR TITLE
added redis and redis-namespace - delivers #37

### PR DESCRIPTION
**Pull request by saneshark**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -28,6 +28,10 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 # Discussion: https://github.com/18F/college-choice/issues/597#issuecomment-139034834
 gem "puma", "~> 2.16.0"
 
+#redis and redis-namespace for session and mvi persistence
+gem 'redis'
```
Comment (robbiethegeek) [inline Gemfile]: can we add a version that maps to the redis version that is supported by GovCloud
from docs : Redis Version 2.8.24
http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/SelectEngine.html

File: README.md [hunk 2]
```diff
@@ -4,11 +4,23 @@ This project provides common APIs for applications that live on vets.gov. This r
...
+
 1. Install Ruby 2.3. (It is suggested to use a Ruby version manager such as [rbenv](https://github.com/rbenv/rbenv#installation) and then to [install Ruby 2.3](https://github.com/rbenv/rbenv#installing-ruby-versions)).
 1. Install Bundler to manage dependencies: `gem install bundler`
 1. Setup the database: `bundle exec rake db:migrate`
 1. Start the application: `bundle exec rails s`
 
+### Redis
+
+1. Install Redis (on mac): `brew install redis`
+1. Follow post install instructions a) always have redis running as service, b) manually launch: `redis-server /usr/local/etc/redis.conf`
```
Comment (robbiethegeek) [inline README.md]: let's make sure that this maps to the govcloud supported version specifically
Redis Version 2.8.24
http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/SelectEngine.html

File: Gemfile (hunk 1 above)
Comment (saneshark) [inline Gemfile]: This is just the redis gem though, not Redis. According to docs, the latest version supports Redis 2.0+

File: README.md (hunk 2 above)
Comment (saneshark) [inline README.md]: If we would like local version of Redis that developer installs to be fixed to that, we might need to install via homebrew cask / tap, to get the older version. But I'm thinking that this is probably not necessary for local install, just going to be a deployment thing. Later versions of Redis should still be compatible with old, and we can try identify any issues when we have a staging environment.

I could put a note in the deployment section of the readme to call attention to this though if you would like.

Comment (robbiethegeek) [inline README.md]: I think that is ok, I am worried about having things work / available in newer versions of redis on local development environments then is available in GovCloud.  I think I am not going to be happy til we have a docker compose solution and we can specify the versions of Ruby once (not rbenv or rvm) and Redis to lock things down, and limit the works on my machine issues.

File: Gemfile (hunk 1 above)
Comment (robbiethegeek) [inline Gemfile]: discussed in slack

**Comment by robbiethegeek** [comment]
LGTM

**Pull request merged**